### PR TITLE
Add dishes to menu

### PR DESCRIPTION
**Pull request by benjamincomley**
## Add dishes to menu
https://www.pivotaltracker.com/story/show/145100539

- Add dish model
- Add dishes to the correct menu and restaurant
- List dishes in the correct menu together with price
- Create the following dish attributes: Name, Description, Price

### CODE REVIEW

File: features/step_definitions/dish_steps.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Given(/^the following dishes exist$/) do |table|
+  table.hashes.each do |hash|
+    rest = Restaurant.find_by(name: hash["restaurant"])
+    hash["menu"] = Menu.where(name: hash["menu"], restaurant_id: rest.id).first
+    hash["price"] = hash["price"].to_i
+    restaurant = hash.delete("restaurant")
+    FactoryGirl.create(:dish, hash)
+  end
```
Comment (tochman) [inline features/step_definitions/dish_steps.rb]: Not sure about this block. Can you try to explain each line? What is does and why?

File: spec/factories/dishes.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FactoryGirl.define do
+  factory :dish do
+    name "MyString"
+    description "MyString"
+    price 10
+    menu
```
Comment (tochman) [inline spec/factories/dishes.rb]: I would like you to make sure that the dish you create with this factory actually belongs to a menu.

File: spec/models/dish_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  describe 'DB table Dish' do
+    it { is_expected.to have_db_column :id }
+    it { is_expected.to have_db_column :name }
+    it { is_expected.to have_db_column :description }
+    it { is_expected.to have_db_column :price }
+  end
+
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :name }
+    it { is_expected.to validate_presence_of :price }
```
Comment (tochman) [inline spec/models/dish_spec.rb:13]: I assume that we don't validate that a dish actually belongs to a menu. That surprises me a bit. Is there a use case where there would be a need for a dish that is NOT associated with a menu?

**Review by tochman — CHANGES_REQUESTED** [review]

File: features/step_definitions/dish_steps.rb (hunk 1 above)
Comment (benjamincomley) [reply features/step_definitions/dish_steps.rb]: Line 6 now reads `hash.delete("restaurant")`

Line 5. Price value in the hash is converted to an integer as the dish model requires the price to be an integer

Line 6. Restauraunt key value is deleted form the hash as the restaurant is only used to make sure that we're getting the correct menu, as searched for on Line 4 in the code (several restaurants could have a breakfast menu). After we've got the correct menu we need to delete the restaurant from the hash before we can call Factory Girl.

Line 7. Factory Girl creates a Dish from the values given in the hash

File: spec/factories/dishes.rb (hunk 2 above)
Comment (benjamincomley) [reply spec/factories/dishes.rb]: Updated Row 6 to `association :menu`
Hopefully this is what you're looking for?

File: spec/models/dish_spec.rb (hunk 3 above)
Comment (benjamincomley) [reply spec/models/dish_spec.rb:13]: We think this is being checked on Lines 16 through 18?

Comment (tochman) [reply spec/models/dish_spec.rb:13]: Yes. I must have missed these specs. Sorry

**Review by tochman — APPROVED** [review]

**Pull request merged**